paasta_tools/utils.py

```diff
@@ -3076,20 +3078,28 @@ def read_service_instance_names(
     service: str, instance_type: str, cluster: str, soa_dir: str
 ) -> Collection[Tuple[str, str]]:
     instance_list = []
-    conf_file = f"{instance_type}-{cluster}"
+    if instance_type == "spark":
+        conf_file = f"tron-{cluster}"
+    else:
+        conf_file = f"{instance_type}-{cluster}"
     config = service_configuration_lib.read_extra_service_information(
         service,
         conf_file,
         soa_dir=soa_dir,
         deepcopy=False,
     )
     config = filter_templates_from_config(config)
-    if instance_type == "tron":
+    if instance_type == "tron" or instance_type == "spark":
         for job_name, job in config.items():
-            action_names = list(job.get("actions", {}).keys())
-            for name in action_names:
-                instance = f"{job_name}.{name}"
-                instance_list.append((service, instance))
+            for action, action_info in job.get("actions", {}).items():
+                instance = f"{job_name}.{action}"
+                if (
+                    action_info.get("executor", "") == "ssh"
+                    and instance_type == "spark"
+                ):
+                    instance_list.append((service, instance))
+                elif instance_type == "tron":
```

Comment:
this could just be an `else`, right?

Reply:
I can add `else`. Reason I added `elif` to ignore other cases. Example, I don't want to add a case when instance_type is `spark` but executor is not `ssh` or the command is not `spark-submit. 
I think since the action is same both conditions I will make it a single statement.